### DISCUSSION TITLE
Using relative paths more often

### DISCUSSION THREAD

**Discussion by ldelossa** [nvim-mini/mini.nvim#1038] [Q&A]
Hello, 

I use mini.nvim files and picker to open files.
I also use picker to list buffers.

Often, fully-qualified paths are used, instead of relative paths. 

For example:
![image](https://github.com/echasnovski/mini.nvim/assets/5642902/3299a615-56c8-493d-9b89-17603e602889)

Is there a way to change this and use relative paths across these plugins?

**Comment by echasnovski** [comment]
If you mean `:Pick files`, then its items are computed with [configured CLI tool](https://github.com/echasnovski/mini.nvim/blob/072ef026a538cf4a9dedba56ad8eecffa9e9abb6/doc/mini-pick.txt#L1055-L1066) (`rg` by default) with possible fallback to a built-in "fallback" method. For me with `rg` it shows paths relative to current working directory (which you can see with `:pwd`).

If you mean `:Pick buffers`, then it uses the output of a `:buffers` command, which also tries to use relative paths.

So it already does what you're asking for. Maybe you have current working directory not the one you expect (check with `:pwd`)?

**Reply by ldelossa** [reply]
Now that I'm exploring this a bit more, it seems the fully qualified buffer names only occur when I use mini.starter's recent file to open the buffer, or when I'm using `Pick lsp` to navigate to code. 


https://github.com/echasnovski/mini.nvim/assets/5642902/d12f4218-72c9-4ec9-b4a2-59a03269b38b

In the above screen cap I
1. Use vim's edit command to open the most recent file, buffers shows relative file
2. I use starter's recent file list to open the most recent file, fully qualified file winds up in buffer list
3. I use `Pick lsp` via a keyboard shortcut, newly opened buffer has fully qualified file name 
4. I close vim and re-open and repeat 1 for demonstration purposes. 

Don't these plugins need to provide the relative path when opening the buffer to preserve this? My assumption is that the plugins are opening the buffers as fully qualified?

**Reply by echasnovski** [reply]
Yes, this is the issue of 'mini.starter' and (possible) `MiniExtra.pickers.lsp()`. Would you mind opening a single feature request for that?

**Reply by ldelossa** [reply]
Yup, no problem.

**Reply by ldelossa** [reply]
https://github.com/echasnovski/mini.nvim/issues/1044 here you go.
